contrib/msggen/msggen/rust.py

```diff
@@ -0,0 +1,242 @@
+from typing import TextIO
+from typing import Tuple
+from textwrap import dedent, indent
+import logging
+import sys
+
+from .model import (ArrayField, CompositeField, EnumField,
+                    PrimitiveField, Service)
+
+logger = logging.getLogger(__name__)
+
+# The following words need to be changed, otherwise they'd clash with
+# built-in keywords.
+keywords = ["in", "type"]
+
+# Manual overrides for some of the auto-generated types for paths
+# Manual overrides for some of the auto-generated types for paths
+overrides = {
+    'ListPeers.peers[].channels[].state_changes[].old_state': "ChannelState",
+    'ListPeers.peers[].channels[].state_changes[].new_state': "ChannelState",
+    'ListPeers.peers[].channels[].state_changes[].cause': "ChannelStateChangeCause",
+    'ListPeers.peers[].channels[].opener': "ChannelSide",
+    'ListPeers.peers[].channels[].closer': "ChannelSide",
+    'ListPeers.peers[].channels[].features[]': "string",
+    'ListFunds.channels[].state': 'ChannelState',
+}
+
+# A map of schema type to rust primitive types.
+typemap = {
+    'boolean': 'bool',
+    'hex': 'String',
+    'msat': 'Amount',
+    'number': 'i64',
+    'pubkey': 'String',
+    'short_channel_id': 'String',
+    'signature': 'String',
+    'string': 'String',
+    'txid': 'String',
+}
+
+header = f"""#![allow(non_camel_case_types)]
+//! This file was automatically generated using the following command:
+//!
+//! ```bash
+//! {" ".join(sys.argv)}
+//! ```
+//!
+//! Do not edit this file, it'll be overwritten. Rather edit the schema that
+//! this file was generated from
+
+"""
+
+
+def normalize_varname(field):
+    """Make sure that the variable name of this field is valid.
+    """
+    # Dashes are not valid names
+    field.path = field.path.replace("-", "_")
```

Comment:
```suggestion
    field.path = field.path.replace("-", "_").replace("[]", "")
```

Maybe this is the cause of the serde alias name?